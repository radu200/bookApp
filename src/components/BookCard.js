import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';

const useStyles = makeStyles({
  root: {
    minWidth: 275,
  },
  title: {
    fontSize: 14,
  },
  pos: {
    marginBottom: 12,
  },
});

export default function BookCard(props) {
  const classes = useStyles();
  const {
    book_author,
    book_pages,
    book_publication_city,
    book_publication_country,
    book_publication_year,
    book_title
  } = props;

  return (
    <Card className={classes.root} variant="outlined">
      <CardContent>
        <Typography noWrap className={classes.title} color="textSecondary" gutterBottom>
          {book_title}
        </Typography>
        <Typography variant="h5" component="h2">
        </Typography>
        <Typography noWrap className={classes.pos} color="textSecondary">
          {book_author}
        </Typography>
        <Typography variant="body2" component="p">
          {book_publication_country}, {book_publication_city}, {book_publication_year}
          <br />
          {book_pages}
        </Typography>
      </CardContent>
    </Card>
  );
};

BookCard.propTypes = {
  book_author: PropTypes.array,
  book_pages: PropTypes.number,
  book_publication_city: PropTypes.string,
  book_publication_country: PropTypes.string,
  book_publication_year: PropTypes.number,
  book_title: PropTypes.string,
};