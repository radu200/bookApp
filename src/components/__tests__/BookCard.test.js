import { render, cleanup } from '@testing-library/react';
import BookCard from '../BookCard';
afterEach(cleanup);

describe("BookCard tests", () => {
    it('should render correctly with given props', () => {
        const data = {
            book_author: ["author"],
            book_pages: 20,
            book_publication_city: "London",
            book_publication_country: "UK",
            book_publication_year: 2021,
            book_title: "title"
        };
        const { asFragment } = render(<BookCard {...data} />);
        expect(asFragment()).toMatchSnapshot();
    });
});