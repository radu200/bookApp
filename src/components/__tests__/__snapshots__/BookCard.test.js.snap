// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookCard tests should render correctly with given props 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-outlined MuiPaper-rounded"
  >
    <div
      class="MuiCardContent-root"
    >
      <p
        class="MuiTypography-root makeStyles-title-2 MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-noWrap MuiTypography-gutterBottom"
      >
        title
      </p>
      <h2
        class="MuiTypography-root MuiTypography-h5"
      />
      <p
        class="MuiTypography-root makeStyles-pos-3 MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-noWrap"
      >
        author
      </p>
      <p
        class="MuiTypography-root MuiTypography-body2"
      >
        UK, London, 2021
        <br />
        20
      </p>
    </div>
  </div>
</DocumentFragment>
`;
